woocommerce/woocommerce-ios
# Account creation: Optimizations with separate fields and navigation to login if possible

Closes #9407 


## Description

This PR adds optimization to the account creation flow following the latest updates in pecCkj-wI-p2:
- Separates fields email and password during account creation. Since the two screens are almost identical, I reuse the same view and view model, with the new parameter `field` to decide which field to show.
- Handles existing email error by navigating to the login flow. Please also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/767 to see the changes in the WordPressAuthenticator library.
- Changes some wording: main CTA button, the title of the password screen, and moves the Log In button to the navigation bar.
- Changes the title of the Reset Password button in the Login flow to Forgot your password?

## Testing instructions

- On the prologue screen tap Get Started.
- Enter the email address of an existing email and tap Continue. Notice that the login flow is presented with the input email already populated.
- Navigate back to the account creation screen and change the email to a new one. Tap Continue and notice another screen is displayed with the password field only.
- Enter a valid password and tap Continue. A new account should be created successfully.
- Please feel free to test with invalid input to see if the error cases are still handled correctly.

## Screenshots


https://user-images.githubusercontent.com/5533851/232018749-ea238227-ea15-470f-b87b-e5ea74129414.mp4



---
